### PR TITLE
Add support for keywords tag in GPX root

### PR DESCRIPTION
**Pull request by felixLam**
http://www.topografix.com/gpx_manual.asp#keywords

### CODE REVIEW

File: GPX/GPXRoot.h [hunk 1]
```diff
@@ -39,6 +39,9 @@
 /** Metadata about the file. */
 @property (strong, nonatomic) GPXMetadata *metadata;
 
+/** Keywords for the GPX file. */
```
Comment (shagedorn) [inline GPX/GPXRoot.h]: Any recommendation as to how they should be delimited? Should we rather pass an array, and concat using a known delimiter internally?

File: GPX/GPXRoot.m [hunk 2]
```diff
@@ -40,6 +41,7 @@ - (id)init
     if (self) {
         _version = @"1.1";
         _creator = @"http://gpxframework.com";
+        _keywords = @"";
```
Comment (shagedorn) [inline GPX/GPXRoot.m]: Should we rather leave this to `nil`?

File: GPX/GPXRoot.m [hunk 3]
```diff
@@ -53,6 +55,7 @@ - (id)initWithXMLElement:(GPXXMLElement *)element parent:(GPXElement *)parent
     if (self) {
         _version = [self valueOfAttributeNamed:@"version" xmlElement:element required:YES];
         _creator = [self valueOfAttributeNamed:@"creator" xmlElement:element required:YES];
+        _keywords = [self textForSingleChildElementNamed:@"keywords" xmlElement:element];
```
Comment (shagedorn) [inline GPX/GPXRoot.m]: Does this return `nil` or an empty string when the child element is missing?

**Review by shagedorn — COMMENTED** [review]
This does not write out keywords, does it?

File: GPX/GPXRoot.h (hunk 1 above)
Comment (felixLam) [reply GPX/GPXRoot.h]: The manual says: 

> Keywords for indexing the GPX file with search engines. Comma separated.

Will do the parsing here then and make it nullable to differentiate between empty keywords and missing keywords.

File: GPX/GPXRoot.h [hunk 4]
```diff
@@ -37,7 +38,10 @@
 @property (strong, nonatomic) NSString *creator;
```
Comment (shagedorn) [inline GPX/GPXRoot.h]: Mark all remaining public properties `readonly`? Or can this actually be set?

File: GPX/GPXRoot.m [hunk 5]
```diff
@@ -93,6 +96,24 @@ + (GPXRoot *)rootWithCreator:(NSString *)creator;
     return root;
 }
 
++ (NSArray<NSString *> *)keywordsArrayFromString:(NSString *)keywordString {
+    if (!keywordString) {
+        // return nil only for nil strings, to differentiate between empty and nil keywords
+        return nil;
+    }
+
+    NSArray *keywords = [keywordString componentsSeparatedByString:@","];
```
Comment (shagedorn) [inline GPX/GPXRoot.m:108]: If there's only one string (without comma), the array contains that one string, correct?

**Review by shagedorn — APPROVED** [review]

File: GPX/GPXRoot.h (hunk 4 above)
Comment (felixLam) [reply GPX/GPXRoot.h]: I wanted to change as little as possible as part of this PR. We can probably mark them all readonly, as we don't use them. Given that this is a parser, there should be no use in setting different values externally.

File: GPX/GPXRoot.m (hunk 5 above)
Comment (felixLam) [reply GPX/GPXRoot.m:108]: Correct

Comment (felixLam) [reply GPX/GPXRoot.m:108]: https://developer.apple.com/documentation/foundation/nsstring/1413214-componentsseparatedbystring 

> If list has no separators—for example, "Karin"—the array contains the string itself

**Pull request merged**